org-roam.el

```diff
@@ -159,29 +160,77 @@ If called interactively, then PARENTS is non-nil."
         (make-directory paren-dir parents)))
     (write-region "" nil filename nil 0)))
 
-;;; Dynamic variables
-(defvar org-roam-cache-initialized nil
-  "Boolean value indicating whether the cache is initialized.")
-
-(defvar org-roam-forward-links-cache (make-hash-table :test #'equal)
-  "Cache containing forward links.")
-
-(defvar org-roam-backward-links-cache (make-hash-table :test #'equal)
-  "Cache containing backward-links.")
+;;; Classes
 
-(defvar org-roam-titles-cache (make-hash-table :test #'equal)
-  "Cache containing titles for org-roam files.")
+(defclass org-roam-cache ()
+  ((initialized :initarg :initialized
+                :documentation "Is cache valid?")
+   (forward-links :initarg :forward-links
+                  :documentation "Cache containing forward links.")
+   (backward-links :initarg :backward-links
+                   :documentation "Cache containing backward links.")
+   (titles :initarg :titles
+           :documentation "Cache containing titles for org-roam files."))
+  "All cache for an org-roam directory.")
 
+;;; Dynamic variables
 (defvar org-roam--current-buffer nil
   "Currently displayed file in `org-roam' buffer.")
 
 (defvar org-roam-last-window nil
   "Last window `org-roam' was called from.")
 
+(defvar org-roam--cache nil
+  "The list of cache separated by directory.")
+
 ;;; Utilities
+(defun org-roam-directory-normalized ()
+  "Get the org-roam-directory normalized so that it can be used
+as a unique key."
+  (s-chop-suffix "/" (file-truename org-roam-directory)))
```

Comment:
This might flake on windows, which uses forward slashes for directories. Use `directory-file-name`?